### PR TITLE
feat: Add zod data validation to openapi

### PR DESCRIPTION
**Pull request by kainpets**

### CODE REVIEW

File: src/lib/blueprint.ts [hunk 1]
```diff
@@ -403,40 +405,47 @@ const createResponse = (responses: OpenapiOperation['responses']): Response => {
   }
 }
 
+const PropertySchema = z
```
Comment (razor-x) [inline src/lib/blueprint.ts:408]: I think we should only use zod for the openapi spec parsing and leave blueprint as a normal TypeScript def. Zod is a parse lib, and we only need to parse and normalize the openapi spec. We can set the defaults for missing stuff there.

File: src/lib/openapi.ts [hunk 2]
```diff
@@ -73,10 +77,57 @@ export interface OpenapiSchema {
...
+  'x-deprecated'?: string
 }
 
 export interface OpenapiComponents {
   schemas: Record<string, OpenapiSchema>
 }
 
 export type OpenapiSecurity = Record<string, string[]>
+
+export const openapiOperationSchemaValidator = z.object({
```
Comment (razor-x) [inline src/lib/openapi.ts:91]:
```suggestion
export const Openapi = z.object({
```

File: src/lib/openapi-validator.ts
Comment (razor-x) [inline src/lib/openapi-validator.ts:1]: What is the difference between the validator and the schema? I would expect the validation and the schema is the same thing.

File: src/lib/openapi.ts [hunk 3]
```diff
@@ -1,82 +1,132 @@
...
+  content: z
+    .record(
+      z.object({
+        schema: ZSchema,
+      }),
+    )
+    .optional(),
+})
+
+export const ZOpenapiOperation = z.object({
```
Comment (razor-x) [inline src/lib/openapi.ts:85]: The `Z` prefix is painful. Should we just set this eslint rule `@typescript-eslint/no-redeclare` to `ignoreDeclarationMerge` so we can name the schema and the type the same per the Zod docs? We don't have this issue in most other repos, but things would be much simpler if we allowed it here. 

I have the same issue in my other PR, going to try it there.

Comment (razor-x) [reply src/lib/openapi.ts:85]: Ok, nevermind, using the same name doesn't work. Instead of the Z prefix, how about `Schema` suffix?

Comment (razor-x) [reply src/lib/openapi.ts:85]: I've followed this convention in https://github.com/seamapi/blueprint/pull/26

File: src/lib/openapi-validator.ts
Comment (kainpets) [reply src/lib/openapi-validator.ts:1]: Yeah, it is.

File: src/lib/openapi.ts (hunk 3 above)
Comment (kainpets) [reply src/lib/openapi.ts:85]: I wanted to avoid the atrocity of `schemaSchema` but maybe something like `modelSchema`?

File: src/lib/openapi-validator.ts
Comment (kainpets) [reply src/lib/openapi-validator.ts:1]: I've changed the name because I wanted a more granular approach and I had to make openapi schema recursive

**Comment by kainpets** [comment]
Partailly implemented in another PR #31

File: src/lib/openapi.ts (hunk 3 above)
Comment (razor-x) [reply src/lib/openapi.ts]: Just wanted to note that while having one instance of `OpenapiSchemaSchema` is not ideal, it shouldn't mean the convention is no good.

**Pull request closed without merging**